subprojects/scala/src/main/java/org/gradle/api/tasks/ScalaRuntime.java

```diff
@@ -99,16 +99,18 @@ public FileCollection createDelegate() {
                     throw new AssertionError(String.format("Unexpectedly failed to parse version of Scala Jar file: %s in %s", scalaLibraryJar, project));
                 }
 
+                String zincVersion = project.getExtensions().getByType(ScalaPluginExtension.class).getZincVersion().get();
```

Comment:
To break this cycle, we could also change calls to `inferScalaClasspath` to take another parameter, but that may not really be worth it since this is a public API. :/

All of this looks kind of weird (sniffing the classpath + guessing what the dependencies should be).  It would be nicer if we could go the other way (specify the version and we'll pick everything) like we're doing with Zinc now.

All that being said, this is OK.

Reply:
And I remember seeing tickets that use that public API, so indeed most likely not worth it.